include/bluetooth/gatt.h

```diff
@@ -703,13 +703,23 @@ struct _bt_gatt_ccc {
 	/** Configuration for each connection */
 	struct bt_gatt_ccc_cfg cfg[BT_GATT_CCC_MAX];
 
-	/** Highest value of all connected peer's subscriptions */
+	/** Bitwise-or of all connected peer's CCC values */
 	uint16_t value;
 
-	/** @brief CCC attribute changed callback
+	/** @brief React to changes to the @c value field of this struct
 	 *
-	 *  @param attr   The attribute that's changed value
-	 *  @param value  New value
+	 *  Use this event to optimize when to broadcast using bt_gatt_indicate()
+	 *  and bt_gatt_notify().
+	 *
+	 *  This event will not trigger if only the @c cfg field changes but the
+	 *  @c value field remains the same. Combine the use of this event and
+	 *  @c cfg_write to catch all changes to the @c cfg field.
```

Comment:
I would like to change this to trigger on any change to the `cfg` field. RFC

Reply:
On the other hand, it would be awkward because only `value` is passed as a parameter here.

Reply:
I see your point. 
Technically a device keeps a CCCD for each client which is of course abstracted away by the API, so it appears as just a single value. 

It does sound a bit odd to call the callback every time there is a change, since the value may not actually change here, and the application would thus not really be able to use that information for anything. 

I think in the case where an application needs/wants more specific knowledge or handling of CCCD writes, it would just just the `BT_GATT_CCC_MANAGED` to do that.